Intel icx + -fvisibility=hidden hides weak MPI_* aliases from libmpi.so

While testing MPICH with Intel oneAPI 2026 compilers, the default shared-library build failed when linking MPICH’s own src/env/mpichversion and src/env/mpivars utilities.

This looks like an interaction between Intel icx, MPICH weak-symbol profiling aliases, and --enable-visibility / -fvisibility=hidden.

# Reproducer

```
  ./autogen.sh

  mkdir build-intel-ch4-ofi-g
  cd build-intel-ch4-ofi-g

  source /opt/intel/oneapi/setvars.sh

  CC=icx CXX=icpx FC=ifx F77=ifx ../configure \
    --prefix=$PWD/../install-intel-ch4-ofi-g \
    --with-device=ch4:ofi \
    --with-libfabric=embedded \
    --enable-g \
    --enable-f77 \
    --enable-f90 \
    --enable-f08

  make -j50
```

# Environment

  MPICH commit: 292d45079b11b05241add7f09e7234c69d2a9899
  icx/icpx: Intel oneAPI DPC++/C++ Compiler 2026.0.0 20260331
  ifx: 2026.0.0 20260331
  OS: Linux x86_64, Ubuntu kernel 5.15.0-1104-nvidia

# Configure output of interest
```
  checking for type of weak symbol alias support... pragma weak alias
  checking for multiple weak symbol alias support...
  checking whether to enable symbol visibility... yes (via -fvisibility=hidden)
```
# Failure

  The build reaches lib/libmpi.la, then fails linking utilities against the just-built shared library:
```
  CCLD     src/env/mpichversion
  CCLD     src/env/mpivars
  /usr/bin/ld: src/env/mpichversion.o: undefined reference to `MPI_Get_library_version'
  /usr/bin/ld: src/env/mpivars.o: undefined reference to `MPI_Init_thread'
  /usr/bin/ld: src/env/mpivars.o: undefined reference to `MPI_T_init_thread'
  ...
  icx: error: linker command failed with exit code 1
```

# Verbose link line:
```
  libtool: link: icx -fvisibility=hidden -g -o src/env/.libs/mpichversion \
    src/env/mpichversion.o lib/.libs/libmpi.so \
    -Wl,-rpath -Wl,.../install-intel-ch4-ofi-g/lib
```
# Symbol evidence

  With Intel icx, the shared library exports PMPI_* but not the public weak MPI_* aliases:
```
  $ nm -D build-intel-ch4-ofi-g/lib/.libs/libmpi.so | rg 'MPI_Get_library_version|PMPI_Get_library_version|MPI_Init_thread|PMPI_Init_thread'
  0000000000161b60 T PMPI_Get_library_version
  000000000015eb40 T PMPI_Init_thread
```
The static archive does contain the expected weak aliases:
```
  $ nm -g build-intel-ch4-ofi-g/lib/.libs/libmpi.a | rg 'MPI_Get_library_version|PMPI_Get_library_version|MPI_Init_thread|PMPI_Init_thread'
  0000000000000070 W MPI_Init_thread
  0000000000000070 T PMPI_Init_thread
  0000000000000070 W MPI_Get_library_version
  0000000000000070 T PMPI_Get_library_version
```

For comparison, the GCC shared-library build exports both:
```
  0000000000139090 W MPI_Get_library_version
  0000000000136820 W MPI_Init_thread
  0000000000139090 T PMPI_Get_library_version
  0000000000136820 T PMPI_Init_thread
```

# Expected behavior

The shared libmpi.so should export the public MPI_* entry points when weak-symbol profiling is enabled, or configure should avoid enabling symbol visibility for compiler/linker combinations where the weak aliases become hidden.

# Likely workaround

A separate Intel configure with `--disable-visibility` gets past the relevant configure checks and should avoid hiding the weak aliases. I am testing that build separately.

--- a/confdb/aclocal_cc.m4
+++ b/confdb/aclocal_cc.m4
@@ -229,6 +229,7 @@ int Weak_Fn_B(int); int Fn_B(int);
 #pragma weak Weak_Fn_B = Fn_B
 int Fn_A(int a) { return a; }
 int Fn_B(int a) { return a; }
+int check(int a) { return Weak_Fn_A(a); }
         ])],[
         AC_LANG_SOURCE([
 int Weak_Fn_A(int);
@@ -244,7 +245,7 @@ int main(int argc, char **argv) { return Weak_Fn_A(0) + Weak_Fn_B(0);}
             int Foo(int a) { return a; }
         ]],[[
             return PFoo(1);
-        ]])],pac_cv_prog_c_weak_symbols="pragma _HP_SECONDARY_DEF")
+        ]])],pac_cv_prog_c_weak_symbols="pragma _HP")
     fi
     dnl -------------------------------------
     if test -z "$pac_cv_prog_c_weak_symbols" ; then
@@ -254,22 +255,7 @@ int main(int argc, char **argv) { return Weak_Fn_A(0) + Weak_Fn_B(0);}
             int Foo(int a) { return a; }
         ]],[[
             return PFoo(1);
-        ]])],pac_cv_prog_c_weak_symbols="pragma _CRI duplicate x as y")
-    fi
-    dnl -------------------------------------
-    if test -z "$pac_cv_prog_c_weak_symbols" ; then 
-        PAC_COMPLINK_IFELSE(
-            [AC_LANG_SOURCE([[
-                int Foo(int a);
-                int PFoo(int) __attribute__ ((weak,alias("Foo")));
-                int Foo(int a) { return a; }
-            ]])],
-            [AC_LANG_PROGRAM([[
-                int PFoo(int);
-            ]],[[
-                return PFoo(1);
-            ]])],
-            pac_cv_prog_c_weak_symbols="attribute alias")
+        ]])],pac_cv_prog_c_weak_symbols="pragma _CRI")
     fi
     dnl -------------------------------------
     if test -z "$pac_cv_prog_c_weak_symbols" ; then 
@@ -1281,7 +1267,7 @@ dnl is not declared in a header, this will fail.  Use a non-static global so
 dnl the compiler does not warn about an unused variable.
 dnl
 dnl Simply calling the function is not enough because C89 compilers allow
-dnl calls to implicitly-defined functions.  Re-declaring a library function
+dnl calls to implicitly-defined functions.  Redeclaring a library function
 dnl with an incompatible prototype is also not sufficient because some
 dnl compilers (notably clang-3.2) only produce a warning in this case.
 dnl

